### PR TITLE
kv: divide-and-conquer approach to splitting

### PR DESCRIPTION
**Pull request by petermattis**
Scatter the replicas after each split to get a balanced state quicker.

### CODE REVIEW

**Comment by petermattis** [comment]
Should probably use this same approach in `ycsb`. Or this might be a good candidate for pulling out into a shared package.

File: kv/main.go [hunk 1]
```diff
@@ -328,15 +329,64 @@ func setupCockroach(parsedURL *url.URL) (database, error) {
...
+					if !ok {
+						break
+					}
+					m := (p.lo + p.hi) / 2
+					split := splitPoints[m]
+					if _, err := db.Exec(`ALTER TABLE test.kv SPLIT AT VALUES ($1)`, split); err != nil {
+						log.Fatal(err)
+					}
+					if _, err := db.Exec(fmt.Sprintf(`ALTER TABLE test.kv SCATTER FROM (%d) TO (%d)`,
+						splitPoints[p.lo], split+1)); err != nil {
```
Comment (a-robinson) [inline kv/main.go]: Minor, but a comment here would make this clearer. I assume the `+1` on `split` is to make it hit both sides of the split, but it took some thinking to understand the `+1`.

**Review by a-robinson — APPROVED** [review]
LGTM

File: kv/main.go (hunk 1 above)
Comment (petermattis) [reply kv/main.go]: Heh, it took some thinking to get that code right too. Done.

File: kv/main.go [hunk 2]
```diff
@@ -328,15 +329,66 @@ func setupCockroach(parsedURL *url.URL) (database, error) {
...
+						}
+					}()
+				}
+			}()
 		}
+
+		for i := 0; i < len(splitPoints); i++ {
+			<-doneCh
+			if (i+1)%1000 == 0 {
+				fmt.Printf("%d splits\n", i+1)
```
Comment (rjnn) [inline kv/main.go:391]: Could stick a `printf` after the loop as well, given that someone may use a split number thats not a multiple of 1000.

**Review by rjnn — APPROVED** [review]
LGTM.

File: kv/main.go (hunk 2 above)
Comment (petermattis) [reply kv/main.go:391]: The purpose is to be a progress indicator. We'll start running the test at that point giving adequate indication of progress.

**Pull request merged**